systems/ros/ros_publisher_system.h

```diff
@@ -0,0 +1,198 @@
+   * be published. Supported TriggerTypes are {kForced, kPeriodic, kPerStep}.
+   * Will throw an error if empty or if unsupported types are provided.
+   *
+   * @param publish_period Period that messages will be published (optional).
+   * publish_period should only be non-zero if one of the publish_triggers is
+   * kPeriodic.
+   */
+  static std::unique_ptr<RosPublisherSystem<RosMessage>> Make(
+      const std::string& topic, ros::NodeHandle* node_handle,
+      double publish_period = 0.0) {
+    return std::make_unique<RosPublisherSystem<RosMessage>>(topic, node_handle,
+      publish_period);
+  }
+
+  /**
+   * A factory method that returns an %RosPublisherSystem that takes
+   * Value<RosMessage> message objects on its sole abstract-valued input port.
+   *
+   * @param[in] topic The ROS topic on which to publish.
+   *
+   * @param node_handle The ROS context.
+   *
+   * @param publish_period Period that messages will be published (optional).
+   * If the publish period is zero, RosPublisherSystem will use per-step
+   * publishing instead; see LeafSystem::DeclarePerStepPublishEvent().
+   */
+  static std::unique_ptr<RosPublisherSystem<RosMessage>> Make(
+      const std::string& topic, ros::NodeHandle* node_handle,
+      const TriggerTypeSet& publish_triggers, double publish_period = 0.0) {
+    return std::make_unique<RosPublisherSystem<RosMessage>>(topic, node_handle,
+      publish_triggers, publish_period);
+  }
+
+  /**
+   * A constructor for an %RosPublisherSystem that takes ROS message objects on
+   * its sole abstract-valued input port. The ROS message type is determined by
+   * the template type.
+   *
+   * @param[in] topic The ROS topic on which to publish.
+   *
+   * @param node_handle The ROS context.
+   *
+   * @param publish_triggers Set of triggers that determine when messages will
+   * be published. Supported TriggerTypes are {kForced, kPeriodic, kPerStep}.
+   * Will throw an error if empty or if unsupported types are provided.
+   *
+   * @param publish_period Period that messages will be published (optional).
+   * publish_period should only be non-zero if one of the publish_triggers is
+   * kPeriodic.
+   */
+  RosPublisherSystem(const std::string& topic, ros::NodeHandle* node_handle,
+      const TriggerTypeSet& publish_triggers, double publish_period = 0.0)
+      : topic_(topic), node_handle_(node_handle) {
+    DRAKE_DEMAND(node_handle_ != nullptr);
+    DRAKE_DEMAND(publish_period >= 0.0);
+    DRAKE_DEMAND(!publish_triggers.empty());
+
+    using drake::systems::TriggerType;
+
+    // Check that publish_triggers does not contain an unsupported trigger.
+    for (const auto& trigger : publish_triggers) {
+      DRAKE_THROW_UNLESS((trigger == TriggerType::kForced) ||
+        (trigger == TriggerType::kPeriodic) ||
+        (trigger == TriggerType::kPerStep));
+    }
+
+    publisher_ = node_handle->advertise<RosMessage>(topic, 5);
```

Comment:
Is it overkill here to double check that publisher_ isn't empty? the [ROS wiki page](http://wiki.ros.org/roscpp/Overview/Publishers%20and%20Subscribers) recommends checking.

Is 5 the right queue length? is that a standard value for implementing a publisher with ROS?

Reply:
-Added a check.
-5 is pretty arbitrary. I added a comment mentioning this. The queue should be small, but there's nothing special about 5.